### ISSUE TITLE
Wipe keys from memory on idle/standby/hibernate/shutdown

### ISSUE THREAD

**Issue by mehrdadn** [dlech/KeeAgent#178]
It would be great if this was an option. Thanks!

**Comment by dlech** [comment]
KeeAgent doesn't actually hold keys in memory. It reads them whenever they are needed and then disposes of them, which should wipe them from memory by writing over it with zeros. So, I don't really see how this feature would add any extra security. Maybe I am missing something in your suggestion?

**Comment by mehrdadn** [comment]
Thanks for the quick reply! I thought it does though? I still have keys listed in this window even when I lock my database:
http://lechnology.com/wp-content/uploads/2014/04/h9yubZwVj7.png

**Comment by dlech** [comment]
That is just the public information about the key, the private part is not in memory. If you are really concerned about idle/standby/hibernate, then you should be locking your KeePass database at this time. There is already an option to remove keys when a database is closed/locked.

**Comment by mehrdadn** [comment]
But why is it that when I SSH with the database locked, it still loads the key if the database had been previously unlocked?

Is this screenshot more convincing? http://imgur.com/a/27u0m

**Comment by mehrdadn** [comment]
I don't know if the behavior has changed recently or something, but I'm on version 0.7.11.0.

**Comment by dlech** [comment]
Hmm.. I guess you are right. At one point in time, KeeAgent worked like I was describing in my previous comments, but I guess that I changed it at some point in time so that you can use keys when the database is locked.

> But why is it that when I SSH with the database locked, it still loads the key if the database had been previously unlocked?

Because you have unchecked the box that says "Remove key from agent when database is locked" for those entries.

**Comment by mehrdadn** [comment]
> Because you have unchecked the box that says "Remove key from agent when database is locked" for those entries.

Yes, so for such entries, can you make it so that the keys are wiped when the system undergoes a power transition? That's the issue...
